maturin: 0.10.4 -> 0.10.6

danieldk opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Changelog:
https://github.com/PyO3/maturin/blob/v0.10.5/Changelog.md#0105---2021-05-21

Also add rustc to the dependencies of maturinBuildHook, otherwise maturin
cannot find the Rust compiler.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

danieldk committed:
maturin: 0.10.4 -> 0.10.5

Changelog:
https://github.com/PyO3/maturin/blob/v0.10.5/Changelog.md#0105---2021-05-21

danieldk committed:
maturinBuildHook: add rustc to deps

maturin 0.10.5 uses rustc -vV to find the host:

https://github.com/PyO3/maturin/commit/e886c85f5a422111f7447541fd896228f6d403f1

We now need to make rustc visible to the hook for maturin to work
properly.

jonringer reviewed (commented):
did you want to fix darwin?

```
ld: library not found for -liconv
```

should just be `pkgs.darwin.libiconv`

danieldk committed:
maturin: fix Darwin build

danieldk commented:
> did you want to fix darwin?

Ah nice, ofborg darwin builds work again?

Should be fixed now...

jonringer commented:
> Ah nice, ofborg darwin builds work again?

I guess grahamc asked for some funding from the foundation to add some darwin builders.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 123867` at 11307c1d run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/c1b32a89ce5897f25efb473cff37fb3a">python38Packages.qiskit-aer</a></li>
    <li><a href="https://gist.github.com/5e8a2f61f74c9566cc85da2ab2cc1d42">python38Packages.qiskit-ibmq-provider</a></li>
    <li><a href="https://gist.github.com/8c5afb0ab1344ecbbd4f5d96f5f85484">python38Packages.qiskit-ignis</a></li>
    <li><a href="https://gist.github.com/a9f085591117de70f1ac7b0b8ad77b74">python38Packages.qiskit-terra</a></li>
    <li><a href="https://gist.github.com/d3d6c00aad0c922bb23d00f6e8ffcd40">python39Packages.qiskit-aer</a></li>
    <li><a href="https://gist.github.com/4b0b95b98568827a09e86c7e18df7e9a">python39Packages.qiskit-ibmq-provider</a></li>
    <li><a href="https://gist.github.com/bdaac9f300b65a016cb6533a55b4ce28">python39Packages.qiskit-terra</a></li>
  </ul>
</details>
<details>
  <summary>9 packages skipped due to time constraints:</summary>
  <ul>
    <li>chia</li>
    <li>python38Packages.johnnycanencrypt</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-aqua</li>
    <li>python38Packages.tumpa</li>
    <li>python39Packages.adblock</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-aqua</li>
    <li>python39Packages.qiskit-ignis</li>
  </ul>
</details>
<details>
  <summary>26 packages built successfully:</summary>
  <ul>
    <li>dyndnsc</li>
    <li>maturin</li>
    <li>python38Packages.adblock</li>
    <li>python38Packages.clvm-rs</li>
    <li>python38Packages.dependency-injector</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.fastdiff</li>
    <li>python38Packages.graphene</li>
    <li>python38Packages.json-logging</li>
    <li>python38Packages.retworkx</li>
    <li>python38Packages.snapshottest</li>
    <li>python38Packages.starlette</li>
    <li>python38Packages.wasmer</li>
    <li>python39Packages.clvm-rs</li>
    <li>python39Packages.dependency-injector</li>
    <li>python39Packages.fastapi</li>
    <li>python39Packages.fastdiff</li>
    <li>python39Packages.graphene</li>
    <li>python39Packages.johnnycanencrypt</li>
    <li>python39Packages.json-logging</li>
    <li>python39Packages.retworkx</li>
    <li>python39Packages.snapshottest</li>
    <li>python39Packages.starlette</li>
    <li>python39Packages.tumpa</li>
    <li>python39Packages.wasmer</li>
    <li>qutebrowser</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

SuperSandro2000 commented on pkgs/development/tools/rust/maturin/default.nix:
```diff
@@ -5,26 +5,27 @@
 
   buildInputs = lib.optional stdenv.isLinux dbus
-    ++ lib.optional stdenv.isDarwin Security;
+    ++ lib.optional stdenv.isDarwin [ Security libiconv ];
```
```suggestion
    ++ lib.optionals stdenv.isDarwin [ Security libiconv ];
```

danieldk replied:
Whoops, fixed.

danieldk commented:
Upstream has release another version, so I have updated this PR to 0.10.6.

danieldk committed:
maturin: 0.10.5 -> 0.10.6

Changelog:
https://github.com/PyO3/maturin/blob/main/Changelog.md#0106---2021-05-21

danieldk commented:
Merging, since dependencies build correctly and all comments have been adressed.
